### DISCUSSION TITLE
Tasmotizer translations

### DISCUSSION THREAD

**Discussion by jziolkowski** [tasmota/tasmotizer#78] [Ideas]
Hello, a question.

Do you see the need to translate Tasmotizer to your local language?
If you vote yes, are you willing to put in the work to provide (and ideally maintain) the localization file for your language?

**Comment by OfferLa** [comment]
Hello, thanks for asking.
My local language is Hebrew.
Personally, I don't need any translation to enjoy Tasmotizer.
If any help needed, I think I can make the language file for Hebrew, if any need comes up.

**Reply by sfromis** [reply]
Even if Tasmotizer (and Tasmota) were to have a translation to my language, I'd still stick to the English-language version.

**Reply by OfferLa** [reply]
Me 2.

> Even if Tasmotizer (and Tasmota) were to have a translation to my language, I'd still stick to the English-language version.

**Reply by ascillato** [reply]
> Personally, I don't need any translation to enjoy Tasmotizer.
> If any help needed, I think I can make the language file, if any need comes up.

I totally agree. I personally don't need it. May could be helpful for other non english speakers. If any help is needed I can make the spanish translation.

**Comment by bovirus** [comment]
Personally I think that have an Italian translation for Italian is a godo plus.

@Sformis
@OfferLa 
As you mentioned tasmota have a translation in your local language.
Then please describe why not have tasmotizer in your local language.

Please don't think if you like or don't like to have tasmotizer in local language.
Please think about what are the negative point to have tasmotizer in your languages.

tasmotizer is powerful application that require a good knowledge of the GUI action.
And most of not English user couldn't understand well the meaning of use of Tasmotizer GUI if they don't know well English.

I believw that if the GUI is localized most of not English user could be more confident with it.

**Reply by sfromis** [reply]
What I mentioned is that Tasmota does **not** have a translation to my language (Danish), and I see no benefit from having Tasmota (or other tools like Tasmotizer) localized to this audience, as I'd see it as unlikely for people tinkering with IoT stuff not to have a decent grasp of English (while they may or may not find it a bit difficult having to express themselves). Of course, I'm aware that the degree of widespread English proficiency can vary greatly in different countries and mother tongues.

The downside of having a translation is "wasting" time and development/maintenance resources on doing it, and how my experience with translations is that they too often makes things **harder** to understand by using "weird" words, which technically are translations, but way outside of what words people use in the real world outside of language-extremists. Or simply that that the translator did not carefully review all the contexts phrases are used in, now or in the future.

I'd not spend any time on doing translations to Danish.

**Comment by Jason2866** [comment]
I am not against localization per se. At the moment the build and release process is manually.
Before doing localized builds the process has to be automated. Preferred way via github actions.
I have not the time to do and maintain extra work.
If "someone" provides a working github actions "yml" file...

**Comment by bovirus** [comment]
The tasmotizer compilation process is currently manual for all steps (compialtion Windows/Linux application).
If you update teh source code and the translations are not updated you can compile the source code without erros with the only limitations that strings not translated are showed in English.
Then the project didn't stop for a missing localization.
It's the same for tasmota project or any other project that use localization (ex atsmota/DDTALL/etc).
This is the flexibility about source code and translations.

**Reply by jziolkowski** [reply]
Yes, I know :) having i18n working any other way would be dumb.

**Comment by bovirus** [comment]
@sformis

If tasmota didn't have a Danish translation why don't cooperate with your knowledge to translate tasmota in Danish?
You should just translate a list of strings.

In the same way why don't cooperate to have a Danish translation for tasmotizer?

What's the neegative point, not only for you but for all Danish users, to have tasmotizer GUI in native Danish language then English?

I beleive that if a translation is not done is a good point to start a translation.
Our job will have as result a benefit for all users of our country.

**Reply by sfromis** [reply]
I've already explained the negative points of doing the translation work, and also why I do not consider it good use of time to do a Danish translation. And doing a good translation is a **lot** more work than just going through a list of strings. Context of string usage matters a lot, and also avoiding translating too many words in an effort to make it "maximally" Danish. Many concepts just do not have common usage outside of using the English words, even though a few people do insist on artificial usage of words made up by some language-nerd, making a translation **less** understandable. 

With the situation in Denmark, I see no positive value in doing a forced translation.

**Comment by bovirus** [comment]
I'm so sorry but I'm not absolute agree with you.
Using your approach no technical sofwtare need a translation in Danish language.
Because most of important functions in a software are not only source code but also a clear and translated GUI that help to use the software as much fine as.
As I explained there is no addition load in source code (except translation) to have a translation working.
For source code just replace one time in .py file string like "Text here" with _("Text Here").
Then if the sofwtare find a translation file with strings use it otherwise use default English.
But no need to add changes in source code for Italian/Danish or other languages. 
Then if you want you can translate it otherwise leave in English.
But for that didn't changes source code.

**Comment by jziolkowski** [comment]
> Using your approach no technical sofwtare need a translation in Danish language.

no, you're not right. You can't take a sentence from a discussion related to a small, easy tool (tasmotizer) and apply it in general to every "technical software" in Danish language.

**Comment by bovirus** [comment]
For me is not related to small or big sofwtare.
But it's related to the importance to understand the use of the sofwtare.
That for a software that make a flash on device for me it's very important that the user knows very well the use of ANY part of the software.
And most the user probably understand well, if is in local language than if it's in English.
Because most of the user are not oblidge to know English.
And probably if the sofwtare is in locale language are encouraged to use it if it's only in English.

**Comment by jziolkowski** [comment]
Yet the idea of tasmotizer is to hide as much details as needed so the user can use it even with very little understanding of english.

Adding localization to application limits my flexibility of changing current widget texts and messages. It requires of me to maintain a base template of these descriptions. Then it requires other to ensure that translations are up to date for every string. That, in turn, will require of me to release only to update strings in some language. All that makes sense in projects that are released often (see tasmota). But after tasmotizer has ESP32 included, it won't see another release pretty much for like half a year, because what else could be added to it.

Either way, my current priorities are to add ESP32 to tasmotizer. After that's done maybe I'll have time to do other stuff.
Also I've read about i18n and I'm more inclined to go with the QTranslator method from QT5 framework, which doesn't require any .bat files to compile/merge anything.

Meanwhile, if you'd like to, you can always fork and distribute tasmotizer-IT to your heart's content.

**Comment by bovirus** [comment]
If it simplify the developing use of QTranslator .TS/.QM file  it's a good option.
Please remind that require also use of LUPDATE to align master to other language.
I hope that at the end of EP32 job you cam consider use of translation (with your way that you like).

**Comment by jziolkowski** [comment]
We'll see. Not going to lie, I would need more users to be interested in translations to justify the added work.

**Comment by ascillato** [comment]
@bovirus 

So far we have just 5 votes. Obviously is not representative for the huge amount of users that love Tasmotizer ![Tasmotizer Github Downloads](https://img.shields.io/github/downloads/tasmota/tasmotizer/total.svg). Just a quick search in YouTube and you will see. So, this issue needs a little more of propaganda so as to gather together people to contribute to Tasmotizer.

**Reply by bovirus** [reply]
@ascillato 

I believe that 5 requests are not a few.
Expeccialy requests from user that is using a github repository that are technical users (not common users).

How can make a better propaganda about feedback for this impovement request?
